### ISSUE TITLE
WS001 false positives

### ISSUE THREAD

**Issue by Daniel-Lecoq** [docToolchain/asciidoc-linter#10]
Hi, nice project! I'm quite new to AsciiDoc but I wanted to implement a pre-commit hook and CI linting so thought I would give this a go. Sorry in advance if I have missed something obvious.

I'm getting WS001 warnings which look to me like false positives. I'm unable to resolve them and I'm looking for a workaround or the option to disable the rule entirely. I've had a look through the docs but so far no success, even with a config yml to disable the rule.

The --verbose and --debug flags are not recognised by the linter, so I'm unable to provide any of that unfortunately.
```
(asciidoc-venv) user:~/server-autopatching$ asciidoc-linter --verbose docs/content/domain-patching-order2.adoc
usage: asciidoc-linter [-h] [--config CONFIG] [--format {console,plain,json,html}] files [files ...]
asciidoc-linter: error: unrecognized arguments: --verbose
```

Adoc file (content removed but issue is with the headings)
```adoc
= JBoss Domain installations

== Order of patching

== Commands

== Server pairs (JBoss Domain)

```

Output (console, no config file)
```
(asciidoc-venv) user:~/server-autopatching$ asciidoc-linter docs/content/domain-patching-order2.adoc
Results for docs/content/domain-patching-order2.adoc:
✗ docs/content/domain-patching-order2.adoc, line 1: Section title should be followed by a blank line
✗ docs/content/domain-patching-order2.adoc, line 2: Section title should be preceded by a blank line
✗ docs/content/domain-patching-order2.adoc, line 2: Section title should be followed by a blank line
✗ docs/content/domain-patching-order2.adoc, line 3: Section title should be preceded by a blank line
✗ docs/content/domain-patching-order2.adoc, line 3: Section title should be followed by a blank line
✗ docs/content/domain-patching-order2.adoc, line 4: Section title should be preceded by a blank line
```

Config file
```yml
# .asciidoc-lint.yml
rules:
  WS001:
    enabled: false
    severity: warning
```

Output (console, with config file)
```
(asciidoc-venv) user:~/server-autopatching$ asciidoc-linter --config .asciidoc-lint.yml docs/content/domain-patching-order2.adoc
Results for docs/content/domain-patching-order2.adoc:
✗ docs/content/domain-patching-order2.adoc, line 1: Section title should be followed by a blank line
✗ docs/content/domain-patching-order2.adoc, line 2: Section title should be preceded by a blank line
✗ docs/content/domain-patching-order2.adoc, line 2: Section title should be followed by a blank line
✗ docs/content/domain-patching-order2.adoc, line 3: Section title should be preceded by a blank line
✗ docs/content/domain-patching-order2.adoc, line 3: Section title should be followed by a blank line
✗ docs/content/domain-patching-order2.adoc, line 4: Section title should be preceded by a blank line
```

JSON Output
```json
{
  "findings": [
    {
      "file": "docs/content/domain-patching-order2.adoc",
      "line": 1,
      "column": null,
      "message": "Section title should be followed by a blank line",
      "severity": "warning",
      "rule_id": "WS001",
      "context": "= JBoss Domain installations"
    },
    {
      "file": "docs/content/domain-patching-order2.adoc",
      "line": 2,
      "column": null,
      "message": "Section title should be preceded by a blank line",
      "severity": "warning",
      "rule_id": "WS001",
      "context": "== Order of patching"
    },
    {
      "file": "docs/content/domain-patching-order2.adoc",
      "line": 2,
      "column": null,
      "message": "Section title should be followed by a blank line",
      "severity": "warning",
      "rule_id": "WS001",
      "context": "== Order of patching"
    },
    {
      "file": "docs/content/domain-patching-order2.adoc",
      "line": 3,
      "column": null,
      "message": "Section title should be preceded by a blank line",
      "severity": "warning",
      "rule_id": "WS001",
      "context": "== Commands"
    },
    {
      "file": "docs/content/domain-patching-order2.adoc",
      "line": 3,
      "column": null,
      "message": "Section title should be followed by a blank line",
      "severity": "warning",
      "rule_id": "WS001",
      "context": "== Commands"
    },
    {
      "file": "docs/content/domain-patching-order2.adoc",
      "line": 4,
      "column": null,
      "message": "Section title should be preceded by a blank line",
      "severity": "warning",
      "rule_id": "WS001",
      "context": "== Server pairs (JBoss Domain)"
    }
  ]
}
```

**Comment by ascheman** [comment]
Having the very same problem, choose any of the README.adoc files of https://github.com/aim42/htmlSanityCheck (or `src/docs/arc42/chapters/*.adoc` ...

**Comment by Daniel-Lecoq** [comment]
@rdmueller I can confirm that the rule disabling works as expected with the above commit. Let me know if I can help with any other verification.

(asciidoc-venv) user@laptop:~/server-autopatching$ asciidoc-linter --config .asciidoc-lint.yml docs/content/domain-patching-order2.adoc
✓ No issues found

**Comment by Daniel-Lecoq** [comment]
I just see the failing black test here https://github.com/docToolchain/asciidoc-linter/actions/runs/13628135951/job/38089741231?pr=12

When running the black formatter locally on your branch against linter.py it reformats single quotes to double quotes on lines 56, 64, 67 and 70.
I'm not familiar with the black formatter so I don't know if formatting differs between environments/installations.

**Comment by rdmueller** [comment]
thanx! So I just learned something new. While the rule can now be deactivated, I think the base problem isn't fixed yet, because it wasn't described in detail in the issue. Can you open another issue which explains, why these are false positives?
